### PR TITLE
rimage: add setting of production bit

### PR DESCRIPTION
**Pull request by juimonen**
When building and signing product images a product bit must be set in manifest headers. Add option -p to rimage to set it.

Signed-off-by: Jaska Uimonen <jaska.uimonen@intel.com>
Co-developed-by:: Adrian Bonislawski <adrian.bonislawski@intel.com>

### CODE REVIEW

File: src/adsp_config.c [hunk 1]
```diff
@@ -1023,8 +1037,10 @@ static int parse_signed_pkg(const toml_table_t *toml, struct parse_ctx *pctx,
 	/* bitmap array */
 	bitmap_array = toml_array_in(signed_pkg, "bitmap");
 	if (!bitmap_array) {
-		/* default value */
-		out->bitmap[4] = 8;
+		if (production)
+			out->bitmap[4] = 0x8;
```
Comment (abonislawski) [inline src/adsp_config.c:1041]: This is not related to prod bit, please revert all bitmap[] changes

**Review by abonislawski — CHANGES_REQUESTED** [review]
@lgirdwood can you confirm that we can expose reserved0 (production) bit functionality to public?

File: src/adsp_config.c (hunk 1 above)
Comment (abonislawski) [reply src/adsp_config.c:1041]: or you can add another commit to setup properly IMR type, there is already such option:

fprintf(stdout, "\t -i set IMR type\n");

so you can set bitmap to 0x10 if imr type = 4

File: src/rimage.c [hunk 2]
```diff
@@ -40,11 +40,12 @@ int main(int argc, char *argv[])
 	const char *adsp_config = NULL;
 	int opt, ret, i, first_non_opt;
 	int imr_type = MAN_DEFAULT_IMR_TYPE;
+	bool prod_bit = false;
 	int use_ext_man = 0;
 
 	memset(&image, 0, sizeof(image));
 
-	while ((opt = getopt(argc, argv, "ho:va:s:k:ri:f:b:ec:y:q:")) != -1) {
+	while ((opt = getopt(argc, argv, "ho:va:s:k:ri:f:b:ec:y:q:p")) != -1) {
```
Comment (lgirdwood) [inline src/rimage.c:48]: Can we not do this in toml i.e. 
```
intel_reserved0 = 0x1234
```

**Review by lgirdwood — COMMENTED** [review]
Lets make this a toml change

**Comment by lgirdwood** [comment]
> @lgirdwood can you confirm that we can expose reserved0 (production) bit functionality to public?

The naming here is all incorrect, this is for setting IMR bits in the Intel reserved manifest area. Please fix.

File: src/rimage.c (hunk 2 above)
Comment (juimonen) [reply src/rimage.c]: ok

File: src/adsp_config.c (hunk 1 above)
Comment (juimonen) [reply src/adsp_config.c]: ok

**Comment by juimonen** [comment]
ver2: all stuff coming from toml.

**Comment by abonislawski** [comment]
So there will be another toml config for prod build?

**Comment by lgirdwood** [comment]
@juimonen @abonislawski still needed or problem solved ?

**Comment by juimonen** [comment]
@lgirdwood I guess Adrian was asking "do we need then another toml file for production builds with rimage", and if we do this this way (as you suggested), we will need another toml or you modify toml on the fly when you build. I think Adrians fear was will this make the build "dirty", which I'm not sure. But this is fine for me at least.

**Comment by lgirdwood** [comment]
@juimonen ok, I see - we cant carry this in the toml as the flags are different. Cmd line option it is then.

**Comment by abonislawski** [comment]
@lgirdwood @juimonen can we revert this to cmd line version and merge for v2.4 prod build?

**Comment by lgirdwood** [comment]
@lyakh can you make this into cmd line version and cherry-pick to v2.4. Thanks

**Comment by lgirdwood** [comment]
@juimonen can we close this since we have cmd line version now ? o is any other work needed in main ?

**Comment by juimonen** [comment]
yes closing
